yash-semantics/src/expansion/initial/arith.rs

```diff
@@ -193,14 +200,20 @@ struct VarEnv<'a> {
 }
 
 impl<'a> yash_arith::Env for VarEnv<'a> {
+    type GetVariableError = UnsetVariable;
     type AssignVariableError = ReadOnlyError;
 
     #[rustfmt::skip]
-    fn get_variable(&self, name: &str) -> Option<&str> {
+    fn get_variable(&self, name: &str) -> Result<Option<&str>, UnsetVariable> {
         if let Some(Variable { value: Some(Scalar(value)), .. }) = self.env.variables.get(name) {
-            Some(value)
+            Ok(Some(value))
         } else {
-            None
+            match self.env.options.get(Unset) {
+                // TODO If the variable exists but is not scalar, UnsetVariable
+                // does not seem to be the right error.
+                Off => Err(UnsetVariable),
```

Comment:
To resolve this to-do, we will need to change `struct UnsetVariable` into an enum:

```rust
enum GetVariableError {
    UnsetVariable,
    NonScalarVariable { assignment_location: Location },
}
```